src/data_mover_threads.c

```diff
@@ -44,11 +59,21 @@ data_mover_threads_do_operation(struct data_mover_threads_op *op)
 		case VDM_OPERATION_MEMCPY: {
 			struct vdm_operation_data_memcpy *mdata
 				= &op->op.memcpy;
-			memcpy(mdata->dest, mdata->src, mdata->n);
+			if (op->data_fns[MEMCPY_FLAGS] != NULL) {
+				memcpy_flags_fn memcpy =
+					op->data_fns[MEMCPY_FLAGS];
+				memcpy(mdata->dest,
```

Comment:
Don't use reserved/known names for variables. It's just confusing. I'm surprised this doesn't generate a warning.